packages/aws_common/lib/src/io/aws_file_platform_io.dart

```diff
@@ -104,4 +106,14 @@ class AWSFilePlatform extends AWSFile {
     // the constructors ensures it won't reach to this point, but just in case
     throw const InvalidFileException();
   }
+
+  @override
+  String? get contentType {
+    final file = _inputFile;
+    if (file != null) {
+      return mime.lookupMimeType(file.path);
+    }
+
+    return super.contentType;
```

Comment:
If `super.contentType` is not `null`, then it's been manually set by the user, right? If that's true, we should not be deducing the content type, we should just be returning this.

Reply:
Same Q for web, too

Reply:
Yea in that case we should return user defined contentType, will update